### PR TITLE
Dice Bombs

### PR DESCRIPTION
**Pull request by lzk228**
## About the PR

After roll, wait value seconds to boom.
Activated on very first roll & cannot be activated twice.

Little explanation how damage spreads.
If dice dropped 10
(not the right values on the pic but its just an explanation)
Screenshot_20240104_200810
At center it will be x10 damage = 100
At sides x5 = 50
At corner x2.5 = 25

If dropped on 20 then big boom (1 value on screenshot = 15 damag)
Screenshot_20240117_002036

If dropped 1 or 2 - you have critical failure and dice just dissapears :trollface: (not as planned but whatever)

## Why / Balance

Cool idea.

## Technical details

Files changed

## Media

[the final result (youtube)](https://youtu.be/TT-097By2MI)

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Dice bombs added to librarian and passenger uplink!

### CODE REVIEW

**Comment by Aisu9** [comment]
please make the dice gain damage depending on the timer

**Comment by lzk228** [comment]
> please make the dice gain damage depending on the timer

i would, but if i do, it will be afwul maybe idk
explosion values need very fine tuning so

**Comment by Just-a-Unity-Dev** [comment]
the higher the roll = the higher the severity imo like dnd

**Comment by GoodWheatley** [comment]
> the higher the roll = the higher the severity imo like dnd

rolling a nat 1 should instagib you

**Comment by K-Dynamic** [comment]
Nat(max) -1: syndie bomb explosion
Nat(max): singulo/tesla

Make dice available to clown too

**Comment by LookyPadooky** [comment]
I agree that rolling a 20 on the d20 should do something insane, it probably won't happen often either because you

1. have to play librarian (one of the least popular jobs)
2. roll syndicate
3. win a one in twenty chance after spending your well earned syndicate bucks on a really risky gimmick item

so please make it have like a syndie bomb tier explosion.......... it would be really funny, plus people would have 20 seconds to react and try to get the bomb somewhere safe anyway so i think it would be funny

**Comment by lzk228** [comment]
Updated pr info.
Only left is to put them into uplink.

**Comment by Dutch-VanDerLinde** [comment]
> Nat(max) -1: syndie bomb explosion Nat(max): singulo/tesla
> 
> Make dice available to clown too

clown already has so much job specific stuff and librarian gets nothing

**Comment by lzk228** [comment]
Ready for review!

**Comment by lzk228** [comment]
https://github.com/space-wizards/space-station-14/assets/124214523/222ac989-433d-4e12-a167-d4d4e3d0ee16

now on critical failure will be critical failure popup

**Comment by BreadyToad** [comment]
> 2024-01-09_14-59-06.mp4 
> now on critical failure will be critical failure popup

Why 2 is considered a critical failure though? Isn't it supposed to be only a nat 1?

**Comment by K-Dynamic** [comment]
For real the clown and mime should also get access to these since they're associated with toys too

**Comment by BreadyToad** [comment]
> For real the clown and mime should also get access to these since they're associated with toys too

Clown and mime already have job specific items, plus dice are much more commonly associated with tabletop games which are much more structured and suited for the librarian than for chaotic duo (plus my boy librarian needs at least something unique to him)

**Comment by K-Dynamic** [comment]
Offtopic but give the syndie librarian:
- bluespace book
- flying book enemy (will either bite or shoot laser)
- fake monkey cubes but in form of figures

**Comment by lzk228** [comment]
> > 2024-01-09_14-59-06.mp4
> > now on critical failure will be critical failure popup
> 
> Why 2 is considered a critical failure though? Isn't it supposed to be only a nat 1?

idk, i just coded and it become like this

**Comment by lzk228** [comment]
Updated PR info. Now will be big boom if 20 dropped.

**Comment by K-Dynamic** [comment]
based

File: Content.Server/Dice/DiceSystem.cs [hunk 1]
```diff
@@ -22,7 +26,38 @@ public override void Roll(EntityUid uid, DiceComponent? die = null)
         var roll = _random.Next(1, die.Sides + 1);
         SetCurrentSide(uid, roll, die);
 
-        _popup.PopupEntity(Loc.GetString("dice-component-on-roll-land", ("die", uid), ("currentSide", die.CurrentValue)), uid);
         _audio.PlayPvs(die.Sound, uid);
+
+        if (die.DiceBomb && !TryComp<ExplodeOnTriggerComponent>(uid, out var explodeTrigger)) // True only on first roll, so cannot be activated twice
```
Comment (TheShuEd) [inline Content.Server/Dice/DiceSystem.cs:31]: I think DiceBomb should be a separate system, not built into an existing DiceSystem. Just make the events you need and subscribe to them in a separate file.

File: Content.Shared/Dice/DiceComponent.cs [hunk 2]
```diff
@@ -34,4 +34,6 @@ public sealed partial class DiceComponent : Component
     [AutoNetworkedField]
     public int CurrentValue { get; set; } = 20;
 
+    [DataField("diceBomb")]
```
Comment (TheShuEd) [inline Content.Shared/Dice/DiceComponent.cs:37]: Also please, separate component

Dice roll logic and explosive dice logic should not be mixed in the same component and system, as they do completely different things.

**Review by TheShuEd — CHANGES_REQUESTED** [review]
I like the unique content for the librarian, but I'm not sure if it makes sense to give that content to greytides in the same way

**Comment by lzk228** [comment]
@TheShuEd ive tried and i cant do it separate system and component.
if i do it in server it cant use events
if i do it in shared it cannot change explosion component because its in server

**Comment by TheShuEd** [comment]
Hmm, then I suggest waiting for more experienced programmers to review

**Comment by AJCM-git** [comment]
I really like this idea for traitor librarians but will close since the requested changes couldn't be fulfilled, feel free to reopen if you find a way.

**Comment by lzk228** [comment]
@AJCM-git i already answered to review and i think someone of expeirenced coders should make other review

**Comment by AJCM-git** [comment]
> @AJCM-git i already answered to review and i think someone of expeirenced coders should make other review

I mean if you end up finding someone to help you out then you can reopen no problem

**Comment by Mangohydra** [comment]
rolling a 1 should stun you for 7 seconds (same as the omega soap)

**Pull request closed without merging**